src/lxd.py

```diff
@@ -0,0 +1,321 @@
+        """Construct the LXD instance representation.
+
+        Args:
+            pylxd_instance: Instance of pylxd.models.Instance for the LXD instance.
+        """
+        self.name = name
+        self._pylxd_instance = pylxd_instance
+        self.files = LxdInstanceFiles(self._pylxd_instance)
+
+    @property
+    def status(self) -> str:
+        """Status of the LXD instance.
+
+        Returns:
+            Status of the LXD instance.
+        """
+        return self._pylxd_instance.status
+
+    def start(self, timeout: int = 30, force: bool = True, wait: bool = False) -> None:
+        """Start the LXD instance.
+
+        Args:
+            timeout: Timeout for starting the LXD instance.
+            force: Whether to force start the LXD instance.
+            wait: Whether to wait until the LXD instance started before returning.
+
+        Raises:
+            LxdException: Unable to start the LXD instance.
+        """
+        try:
+            self._pylxd_instance.start(timeout, force, wait)
+        except pylxd.exceptions.LXDAPIException as err:
+            raise LxdError(f"Unable to start LXD instance {self.name}") from err
+
+    def stop(self, timeout: int = 30, force: bool = True, wait: bool = False) -> None:
+        """Stop the LXD instance.
+
+        Args:
+            timeout: Timeout for stopping the LXD instance.
+            force: Whether to force stop the LXD instance.
+            wait: Whether to wait until the LXD instance stopped before returning.
+
+        Raises:
+            LxdException: Unable to stop the LXD instance.
+        """
+        try:
+            self._pylxd_instance.stop(timeout, force, wait)
+        except pylxd.exceptions.LXDAPIException as err:
+            raise LxdError(f"Unable to stop LXD instance {self.name}") from err
+
+    def execute(self, cmd: list[str], cwd: Optional[str] = None) -> Tuple[int, IO, IO]:
+        """Execute a command within the LXD instance.
+
+        Exceptions are not raise if command execution failed. Caller should check the exit code and
+        stderr for failures.
+
+        Args:
+            cmd: Commands to be executed.
+            cwd: Working directory to execute the commands.
+
+        Returns:
+            Tuple containing the exit code, stdout, stderr.
+        """
+        lxc_cmd = ["/snap/bin/lxc", "exec", self.name]
+        if cwd:
+            lxc_cmd += ["--cwd", cwd]
+
+        lxc_cmd += ["--"] + cmd
+
+        result = secure_run_subprocess(lxc_cmd)
+        return (result.returncode, io.BytesIO(result.stdout), io.BytesIO(result.stderr))
+
+
+class LxdInstances:
+    """LXD instance manager."""
+
+    def __init__(self, pylxd_client: pylxd.Client):
+        """Construct the LXD instance manager.
+
+        Args:
+            pylxd_client: Instance of pylxd.Client.
+        """
+        self._pylxd_client = pylxd_client
+
+    def all(self) -> list[LxdInstance]:
```

Comment:
Should this return a tuple instead of a list?

Reply:
The `all` method returns a list of LXD instances, there could be zero instances or ten instances returned depend on what LXD instance exists at the monment.
Since it is variable length it should be list or iterator.